crates/swc_cli/src/commands/compile.rs

```diff
@@ -84,6 +84,14 @@ pub struct CompileOptions {
     #[clap(group = "input")]
     files: Vec<PathBuf>,
 
+    /// Preserve the file extensions of the input files.
+    #[clap(long, group = "file_extension")]
+    keep_file_extension: bool,
```

Comment:
what's the use case for this?

Reply:
I'm not sure, I add this just because babel has it. https://github.com/babel/babel/pull/6221
may be only `--out-file-extension` is enought?

Reply:
Let's not 1:1 port features unless we are sure if it's needed. 

Do we have usecases for `out-file-extension` as well?

Reply:
to output .mjs file.

For `---output-file-extension`, maybe someone has both .js and .mjs in their codebase and wants swc to keep the output extension?

Reply:
I feel we may hit some breaking changes for this later, but for now having out-file-extension only seems enough to get some feedbacks.

Reply:
removed